### ISSUE TITLE
NotSerializableException when Record references reflection cache

### ISSUE THREAD

**Issue by CNQiYang** [jOOQ/jOOQ#7062]
### Expected behavior and actual behavior:
After creating a new Record and store it,then cannot serialize it.
exception is 

```
java.io.NotSerializableException: java.lang.reflect.Field
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1184)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348)
	at java.util.ArrayList.writeObject(ArrayList.java:762)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1028)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1496)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348)
	at java.util.concurrent.ConcurrentHashMap.writeObject(ConcurrentHashMap.java:1413)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1028)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1496)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348)
	at java.util.concurrent.ConcurrentHashMap.writeObject(ConcurrentHashMap.java:1413)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1028)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1496)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
	at java.io.ObjectOutputStream.defaultWriteObject(ObjectOutputStream.java:441)
	at org.jooq.impl.DefaultConfiguration.writeObject(DefaultConfiguration.java:1066)
```

I found the stored record configuration has `data` property type is `ConcurrentHashMap<Object, Object>.`
keys `"org.jooq.configuration.reflection-cache.get-matching-members" -> " size = 8"` this is `ArrayList<java.lang.reflect.Field>` 
`"org.jooq.configuration.reflection-cache.get-matching-getter" -> " size = 8"` is `ArrayList<java.lang.reflect.Method>` 
not Serializable.
i also see https://github.com/jOOQ/jOOQ/issues/5290.
if i use Record.detach  will work fine.

### Steps to reproduce the problem:

```scala
val record=DSLContext.newRecord(table,pojo)
record.store()
val bos = new ByteArrayOutputStream
val out = new ObjectOutputStream(bos)
out.writeObject(record)
```

### Versions:

- jOOQ:3.9.1
- Java:1.8
- Database (include vendor):Postgresql
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thank you very much for your report. I can easily reproduce this problem. It happens whenever the reflection cache is involved. A workaround would be to turn of reflection caching using `Settings.reflectionCaching == false` (at the obvious price of not having that cache available).

The fix is to simply not serialise the cache.

**Comment by lukaseder** [comment]
This has been fixed for jOOQ 3.11.0. Patches are planned for 3.10.4 (#7065) and 3.9.7 (#7066). 3.10.4 will ship today.

**Comment by lukaseder** [comment]
As will be documented in the manual (#7067), this change will be backwards incompatible under the general assumptions made by jOOQ on serialisation backwards compatibility.
